lib/idris-ide-mode.coffee

```diff
@@ -9,15 +9,16 @@ class IdrisIdeMode extends EventEmitter
   buffer: ''
   idrisBuffers: 0
 
-  constructor: ->
-    pathToIdris = atom.config.get("language-idris.pathToIdris")
-    @process = spawn pathToIdris, ['--ide-mode']
-    @process.on 'error', @error
-    @process.on 'exit', @exited
-    @process.on 'close', @exited
-    @process.on 'disconnect', @exited
+  start: ->
+    if (not @process?) || @process.killed
+      pathToIdris = atom.config.get("language-idris.pathToIdris")
+      @process = spawn pathToIdris, ['--ide-mode']
+      @process.on 'error', @error
+      @process.on 'exit', @exited
+      @process.on 'close', @exited
+      @process.on 'disconnect', @exited
```

Comment:
The `connected` property is a bool that should tell if it's connected right now (at least according to the docs). I use `killed` now to check if the compiler is running, and only start a new instance if it's not.
The events are used if the compiler crashes or is killed by the user.